config/platform.go

```diff
@@ -0,0 +1,106 @@
+package config
+
+import (
+	"bufio"
+	"context"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"go.viam.com/rdk/logging"
+)
+
+var (
+	cudaRegex         = regexp.MustCompile(`Cuda compilation tools, release (\d+)\.`)
+	dpkgVersionRegex  = regexp.MustCompile(`\nVersion: (\d+)\D`)
+	savedPlatformTags []string
+)
+
+// helper to read platform tags for GPU-related system libraries.
+func readGPUTags(tags []string) []string {
+	// this timeout is for all steps in this function.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+	if _, err := exec.LookPath("nvcc"); err == nil {
+		out, err := exec.CommandContext(ctx, "nvcc", "--version").Output()
+		if err != nil {
+			logging.Global().Errorw("error getting cuda version from nvcc. cuda-specific modules may not load", "err", err)
+		}
+		if match := cudaRegex.FindSubmatch(out); match != nil {
+			tags = append(tags, "cuda:true", "cuda_version:"+string(match[1]))
+		} else {
+			logging.Global().Errorw("error parsing `nvcc --version` output. cuda-specific modules may not load")
+		}
+	}
+	if _, err := exec.LookPath("dpkg"); err == nil {
+		out, err := exec.CommandContext(ctx, "dpkg", "-s", "nvidia-jetpack").Output()
+		// note: the error case here will usually mean 'package missing', we don't analyze it.
+		if err == nil {
+			if match := dpkgVersionRegex.FindSubmatch(out); match != nil {
+				tags = append(tags, "jetpack:"+string(match[1]))
```

Comment:
The module I'm working on will have builds for `cuda:11`, `jetpack:5`, and `jetpack:6`. If a board has both CUDA 11 and Jetpack available, which version of the module will be downloaded? I'd like the Jetpack ones to be considered "more specific" than the CUDA ones, but it's unclear if that will happen.

We shouldn't only append the CUDA tag if there is no Jetpack one: I can imagine another module that requires CUDA but has no Jetpack support, and that should still be compatible on a Jetpack-enabled board.

Reply:
nice answer: there's a priority field we have spec'd but not implemented yet

less nice answer: I'm pretty sure these will resolve in the order you upload them :grimacing: